src/js/main.js

```diff
@@ -25,3 +25,27 @@ document.addEventListener("click", (event) => {
 document.querySelector(".site-nav .nav-smallScreen .search-container svg").addEventListener("click", () => {
 	document.querySelector(".site-nav .nav-smallScreen .search-container input").focus();
 });
+
+// Configuration for observer window size/location in viewport.
```

Comment:
The content observer is only needed for `page.njk` and `post.njk` but not any other pages. Using `main.js` for all these code will have this functionality included for all pages, which is unnecessary and inefficient. An idea to improve:

1. Create a new "content-header-observer.js" containing the utility code for this functionality at line 29-46, which will only be included on `page.njk` and `post.njk`;
2. Line 48-51 will be written on `page.njk` and `post.njk` to instantiate `contentHeaderObserver` for specific selectors on these pages.